internal/monitors/logstash/logstash/monitor.go

```diff
@@ -0,0 +1,232 @@
+type Monitor struct {
+	Output        types.Output
+	conf          *Config
+	ctx           context.Context
+	cancel        context.CancelFunc
+	metricTypeMap map[string]datapoint.MetricType
+}
+
+// Configure the monitor and kick off volume metric syncing
+func (m *Monitor) Configure(conf *Config) error {
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+	m.conf = conf
+	m.metricTypeMap = conf.getMetricTypeMap()
+
+	client := &http.Client{
+		Timeout: time.Duration(conf.TimeoutSeconds) * time.Second,
+	}
+
+	scheme := "http"
+	if conf.UseHTTPS {
+		scheme = "https"
+	}
+
+	dims := m.fetchNodeInfo(client, fmt.Sprintf("%s://%s:%d%s", scheme, m.conf.Host, m.conf.Port, nodePath))
+
+	utils.RunOnInterval(m.ctx, func() {
+		var dps []*datapoint.Datapoint
+
+		for prefix, path := range prefixPathMap {
+			dps = append(dps, m.fetchMetrics(client, fmt.Sprintf("%s://%s:%d%s", scheme, m.conf.Host, m.conf.Port, path), prefix, dims)...)
+		}
+		dps = append(dps, m.fetchPipelineMetrics(client, fmt.Sprintf("%s://%s:%d%s", scheme, m.conf.Host, m.conf.Port, pipelinePath), "node.pipelines", dims)...)
+		dps = append(dps, m.fetchPipelineMetrics(client, fmt.Sprintf("%s://%s:%d%s", scheme, m.conf.Host, m.conf.Port, pipelineStatPath), "node.stats.pipelines", dims)...)
+		dps = append(dps, m.fetchPluginMetrics(client, fmt.Sprintf("%s://%s:%d%s", scheme, m.conf.Host, m.conf.Port, pluginPath), "node.plugins", dims)...)
+
+		now := time.Now()
+		for _, dp := range dps {
+			dp.Timestamp = now
+			m.Output.SendDatapoint(dp)
+		}
+
+	}, time.Duration(conf.IntervalSeconds)*time.Second)
+
+	return nil
+}
+
+func (m *Monitor) fetchNodeInfo(client *http.Client, endpoint string) map[string]string {
+	dims := make(map[string]string)
+
+	if nodeJSON, err := getJSON(client, endpoint); err == nil {
+		if nodeID, exists := nodeJSON["id"]; exists {
+			dims["node_id"], _ = nodeID.(string)
+		}
+		if nodeName, exists := nodeJSON["name"]; exists {
+			dims["node_name"], _ = nodeName.(string)
+		}
+	}
+
+	return dims
+}
+
+func (m *Monitor) fetchMetrics(client *http.Client, endpoint string, prefix string, dimensions map[string]string) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+
+	if metricsJSON, err := getJSON(client, endpoint); err == nil {
+		dps = m.extractDatapoints(prefix, metricsJSON, dimensions)
+	}
+
+	return dps
+}
+
+func (m *Monitor) fetchPipelineMetrics(client *http.Client, endpoint string, prefix string, dimensions map[string]string) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+
+	if metricsJSON, err := getJSON(client, endpoint); err == nil {
+		if pipelines, exists := metricsJSON["pipelines"]; exists {
+			if pipelinesObj, isObject := pipelines.(map[string]interface{}); isObject {
+				for pipelineName, pipelineObj := range pipelinesObj {
+					if pipeline, converted := pipelineObj.(map[string]interface{}); converted {
+						dimsClone := make(map[string]string)
+						for dimk, dimv := range dimensions {
```

Comment:
I would invert the logic leading down to here and flatten these six levels of nesting down to one or two.  You can do that by returning on the "bad" paths.